From: Andrea Azzarone <andrea.azzarone@canonical.com>
Date: Tue, 31 Jul 2018 04:44:12 +0100
Subject: [PATCH 17/26] online-accounts: Hide window after adding an online
 account from another app

Some applications (e.g. gnome-calendar, gnome-todo, etc.) use gnome-control-center
to allow the user to add online accounts. This patch makes sure that g-c-c main window
is closed after the user interacted with the new-account dialog.

Bug-Ubuntu: https://launchpad.net/bugs/1754651
Forwarded: no
Last-Update: 2018-03-09
https://gitlab.gnome.org/GNOME/gnome-control-center/issues/16
---
 panels/online-accounts/cc-online-accounts-panel.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/panels/online-accounts/cc-online-accounts-panel.c b/panels/online-accounts/cc-online-accounts-panel.c
index 0da41a7..b9607b8 100644
--- a/panels/online-accounts/cc-online-accounts-panel.c
+++ b/panels/online-accounts/cc-online-accounts-panel.c
@@ -382,9 +382,14 @@ cc_goa_panel_set_property (GObject *object,
             }
 
           if (g_strcmp0 (first_arg, "add") == 0)
-            command_add (CC_GOA_PANEL (object), parameters);
+            {
+              command_add (CC_GOA_PANEL (object), parameters);
+              g_application_quit (g_application_get_default ());
+            }
           else if (first_arg != NULL)
-            select_account_by_id (CC_GOA_PANEL (object), first_arg);
+            {
+              select_account_by_id (CC_GOA_PANEL (object), first_arg);
+            }
 
           return;
         }
